File: EasyMarkets/app/src/main/java/merryweather/com/easymarkets/di/component/AppComponent.java
package merryweather.com.easymarkets.di.component;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Component;
import merryweather.com.easymarkets.App;
import merryweather.com.easymarkets.di.modules.AppModule;
import merryweather.com.easymarkets.di.modules.RepositoryModule;
import merryweather.com.easymarkets.di.qualifier.AppContext;
import merryweather.com.easymarkets.repository.AppartmentRepository;

/**
 * Created by S on 19.04.2018.
 */

@Singleton
@Component(modules = {AppModule.class, RepositoryModule.class})
public interface AppComponent {

    @AppContext
    Context context();


    AppartmentRepository appartmentRepository();

    void inject(final App theApp);
}

File: Nista/app/src/main/java/com/example/nista/di/component/AppComponent.kt
package com.example.nista.di.component

import android.content.Context
import com.example.nista.App
import com.example.nista.api.MovieService
import com.example.nista.repository.MovieRepository
import com.leroymerlin.cds.di.di.modules.RepositoryModule
import com.leroymerlin.cds.di.modules.AppModule
import com.leroymerlin.cds.di.modules.MovieModule
import com.leroymerlin.cds.di.modules.RetrofitModule
import com.leroymerlin.cds.di.qualifier.AppContext
import dagger.Component

import javax.inject.Singleton


@Singleton
@Component(modules = arrayOf(AppModule::class, MovieModule::class, RepositoryModule::class, RetrofitModule::class))
interface AppComponent {

    @AppContext
    fun context(): Context

    fun movieService(): MovieService

    fun movieRepository(): MovieRepository


    fun inject(receiver : App)
}
